app/api/units_api.rb

```diff
@@ -102,7 +102,7 @@ class UnitsApi < Grape::API
         unit.teaching_period = nil
       end
 
-      if unit_parameters.key?(:draft_task_definition_id)
+      if unit_parameters[:draft_task_definition_id]
```

Comment:
Can you adjust this to use `.present?` to better capture what you are asking here?

Reply:
No worries, done!